src/misc/cycle_bases.c

```diff
@@ -0,0 +1,507 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "igraph_cycles.h"
+
+#include "igraph_adjlist.h"
+#include "igraph_components.h"
+#include "igraph_dqueue.h"
+#include "igraph_error.h"
+#include "igraph_interface.h"
+#include "igraph_memory.h"
+
+#include "core/interruption.h"
+#include "misc/order_cycle.h"
+
+/**** Fundamental cycles *****/
+
+/* Computes fundamental cycles for the connected component containing 'start_vid',
+ * and appends them to 'result'.
+ *
+ * 'visited' must be a vector of length vertex_count.
```

Comment:
Is `vertex_count` the number of vertices in the _whole_ graph or in the component containing `start_vid` only?